File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String 
 
class RobotNewsStation(Node): 
    def __init__(self):
        super().__init__("RobotNewsStation") 
        self.get_logger().info("the publisher has started")
        self.publisher_=self.create_publisher(String,"robot_news_1",10)
        self.timer_=self.create_timer(2,self.publish_data)
    def publish_data(self):
        self.publishData = String()
        self.publishData.data = "haloo this is the published data"   
        self.publisher_.publish(self.publishData) 
 
 
def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStation() 
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;
class RobotNewsStatonNode : public rclcpp::Node {
    public:
        RobotNewsStatonNode() : Node("robot_news_staton") {
            publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news_1",10);
            timer_ = this->create_wall_timer(0.5s,std::bind(&RobotNewsStatonNode::publishNews,this));
            RCLCPP_INFO(this->get_logger(),"started");

        }
 
    private:
        void publishNews(){
            auto msg = example_interfaces::msg::String();
            msg.data = std::string("this is the data cpp");
            publisher_ ->publish(msg);
        }
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStatonNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}